File: src/main/java/dev/zprestige/fire/manager/rotationmanager/PacketSendListener.java
package dev.zprestige.fire.manager.rotationmanager;

import dev.zprestige.fire.Main;
import dev.zprestige.fire.event.bus.EventListener;
import dev.zprestige.fire.event.impl.PacketEvent;
import net.minecraft.network.play.client.CPacketPlayer;

public class PacketSendListener extends EventListener<PacketEvent.PacketSendEvent, Object> {

    public PacketSendListener() {
        super(PacketEvent.PacketSendEvent.class);
    }

    @Override
    public void invoke(final Object object) {
        final PacketEvent.PacketSendEvent event = (PacketEvent.PacketSendEvent) object;
        if (event.getPacket() instanceof CPacketPlayer.Rotation) {
            Main.rotationManager.rotationsPerTick.add(System.currentTimeMillis() + 50L);
        }
    }
}

File: src/main/java/dev/zprestige/fire/module/movement/blink/PacketSendListener.kt
package dev.zprestige.fire.module.movement.blink

import dev.zprestige.fire.event.bus.EventListener
import dev.zprestige.fire.event.impl.PacketEvent.PacketSendEvent
import net.minecraft.network.play.client.*

class PacketSendListener(blink: Blink) : EventListener<PacketSendEvent, Blink>(PacketSendEvent::class.java, blink) {
    
    override fun invoke(e: Any) {
        val event = e as PacketSendEvent
        if (module.nullCheck() && !mc.isSingleplayer) {
            val packet = event.packet
            if (module.cPacketPlayer.value && packet is CPacketPlayer) {
                event.setCancelled()
                module.packets.add(packet)
            }
            if (!module.cPacketPlayer.value && !(packet is CPacketChatMessage || packet is CPacketConfirmTeleport || packet is CPacketKeepAlive || packet is CPacketTabComplete || packet is CPacketClientStatus)) {
                module.packets.add(packet)
                event.setCancelled()
            }
        }
    }
}
